lib/validate.test.js

```diff
@@ -166,5 +240,69 @@ describe(__filename, function () {
           done();
         });
     });
+
+    it('should strip the where parameter and accept the list request', function (done) {
+      request
+        .get('/gadgets?who=Sue&where=there')
+        .expect(200)
+        .expect('content-type', /json/)
+        .end(function (err, res) {
+          expect(err).to.not.exist;
+          expect(res.body).to.be.an('array');
+          expect(res.body).to.have.length(0);
+
+          done();
+        });
+    });
+
+    it('should use the default for who and accept the list request', function (done) {
+      request
+        .get('/gadgets?what=123')
+        .expect(200)
+        .expect('content-type', /json/)
+        .end(function (err, res) {
+          expect(err).to.not.exist;
+          expect(res.body).to.be.an('array');
+          expect(res.body).to.have.length(0);
+
+          done();
+        });
+    });
+
+    it('should accept the update request, fire twice?', function (done) {
+      request
+        .put('/gidgets')
+        .set('Accept', 'application/json')
+        .send({
+          number: 123,
+          extraneous: 'who me?'
+        })
+        .expect(200)
+        .expect('content-type', /json/)
+        .end(function (err, res) {
+          expect(err).to.not.exist;
+          expect(res.body).to.be.an('array');
+          expect(res.body).to.have.length(0);
+
+          done();
+        });
+    });
+
+    it('should reject the list request because, convert is set to false', function (done) {
```

Comment:
This is an interesting one, almost like we're testing Joi, but also it verifies our options are respected by Joi. I'm in favour :+1: